src/Servers/Kestrel/Core/src/Internal/Http/HttpRequestHeaders.cs

```diff
@@ -243,11 +243,11 @@ internal Enumerator(HttpRequestHeaders collection)
                 : default;
         }
 
-        public KeyValuePair<string, StringValues> Current => _current;
+        public readonly KeyValuePair<string, StringValues> Current => _current;
 
-        object IEnumerator.Current => _current;
+        readonly object IEnumerator.Current => _current;
 
-        public void Dispose()
+        public readonly void Dispose()
```

Comment:
Is there a benefit to marking Dispose() methods as readonly on Enumerators? I imagine any callers to Dispose() would also be calling MoveNext() anyway. I like reserving the right to mutate this struct in the future on Dispose() even if I don't currently know why we would.

Reply:
I just went through every property and method and marked them as readonly if they didn't modify state.

Even if this was public, I don't believe it is a breaking change to add or remove readonly. Just like it isn't a breaking change on a struct.

No benefit, but no disadvantage either.

Reply:
> For every API touched, we have to evaluate it through the lens of whether we'd ever want to allow mutation to occur -- it would be a breaking change to remove readonly later. For example, adding caching to a readonly method later would be a breaking change.

From @tannergooding in https://github.com/dotnet/runtime/issues/1718. I'm guessing that if readonly is later removed Dispose() and dependencies don't recompile, they wouldn't create a defensive copy of the enumerator when they should.

Reply:
> Even if this was public,

I missed that this isn't public. I guess it's a moot point then.